src/Security/Authentication/Core/src/ISecureDataFormat.cs

```diff
@@ -32,7 +32,7 @@ public interface ISecureDataFormat<TData>
         /// <param name="protectedText">The data protected value.</param>
         /// <returns>An instance of <typeparamref name="TData"/>.</returns>
         [return: MaybeNull]
-        TData Unprotect(string protectedText);
+        TData Unprotect(string? protectedText);
```

Comment:
I think you could do this now:

```suggestion
        TData? Unprotect(string? protectedText);
```

and remove `MaybeNull`